python/hsfs/core/tfdata_engine.py

```diff
@@ -0,0 +1,484 @@
+    pass
+
+try:
+    import boto3
+except ModuleNotFoundError:
+    pass
+
+
+class TFDataEngine:
+    SUPPORTED_TFDTYPES = [
+        tf.float16,
+        tf.float32,
+        tf.float64,
+        tf.int16,
+        tf.int64,
+        tf.int32,
+    ]
+    SPARK_TO_TFDTYPES_MAPPINGS = {
+        "string": tf.string,
+        "short": tf.int16,
+        "int": tf.int32,
+        "long": tf.int64,
+        "float": tf.float32,
+        "double": tf.float64,
+    }
+
+    def __init__(
+        self,
+        training_dataset,
+        split,
+        target_name,
+        feature_names,
+        is_training,
+        cycle_length,
+    ):
+        """TFDataEngine object that has utility methods for tf.data
+
+        # Arguments
+        training_dataset: str, required
+            training dataset name
+        split: split: str, required
+            name of training dataset split. train, test or eval
+        target_name: str, required
+            name of the target variable
+        feature_names: 1d array, required
+            name of training variables
+        is_training: boolean, required
+            whether it is for training, testing or eval
+        cycle_length: int, required
+            number of files to be read and deserialized in parallel.
+
+        # Returns
+            TFDataEngine object
+        """
+
+        self._training_dataset = training_dataset
+        self._split = split
+        self._target_name = target_name
+        self._feature_names = feature_names
+        self._is_training = is_training
+        self._cycle_length = cycle_length
+
+        self._features = self._training_dataset._features
+        self._training_dataset_format = self._training_dataset.data_format
+
+        self._input_files = _get_training_dataset_files(
+            self._training_dataset.location,
+            self._split,
+            filter_empty=True if self._training_dataset_format == "csv" else False,
+        )
+
+        if self._feature_names is None:
+            self._feature_names = [feat.name for feat in self._features]
+
+            if target_name in self._feature_names:
+                self._feature_names.remove(target_name)
+
+    def get_serialized_example_schema(self):
+        """
+        returns schema for parsing serialized example of ff record file
+
+        :return: schema for tfrecord file
+        :rtype: dict
+        """
+        _, tfrecord_feature_description = _get_tfdataset(
+            self._input_files, self._cycle_length
+        )
+        return tfrecord_feature_description
+
+    def tf_record_dataset(
+        self,
+        batch_size=None,
+        num_epochs=None,
+        one_hot_encode_labels=False,
+        num_classes=None,
+        process=False,
+        serialized_ndarray_fname=[],
+    ):
+        """
+        Reads tfrecord files and returns ParallelMapDataset or PrefetchDataset object ready to feed tf keras models
+
+        Example:
+            import hsfs
```

Comment:
no need for the imports etc. in the example I think you can assume the user has a feature store handle already